### PR TITLE
PCHR-2377: Default target contact to current contact on contact details page

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
This PR adds the functionality to display current contact as default target contact, when a new document modal is opened in contact document page.

## Before
- Not default target contact is displayed (empty)
screen shot 2017-08-08 at 3 12 11 pm

## After
- Displayed respective contact as default target contact in new document modal.
2377-after

## Technical Details
1. Load the current contact data and cache for future reference in modules/run.js file.
```js 
var contactsToCache = [config.LOGGED_IN_CONTACT_ID];
config.CONTACT_ID && contactsToCache.push(config.CONTACT_ID);

ContactService.get({ 'IN': contactsToCache }).then(function (data) {
    ContactService.updateCache(data);
});
```

2. Add code to filter cached contacts based on the current contact for target contact lists in "modals/modal-document.js" as below:
```js
function initialContacts (type) {
   ...
   return !$scope.document.id && modalMode === 'edit' ? [] : cachedContacts.filter(function (contact) {
       ...
   });
}
```
3. Changes to adhere new angular js style in modules/run.js
4. Added test case for the changes in modal/modal-document.js in modal-document_test.js

## Comments
- All test files have been renamed to adhere new angular style guide in PR  #239, so the test files in this PR are left as it is now. After #239 is merged it will be rebased to fixed to rename test files as per angularjs style guide.
---

[x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ define([
     '$filter', '$uibModal', '$dialog', '$timeout', 'AssignmentService', 'DocumentService', 'ContactService', 'FileService', 'data',
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js]: Name of the file should be modal-document.controller.js. Please amend this file according to the styleguide as much as possible, including utilisation of $inject, **if possible** getting rid of $scope, private functions + vm.function, order of things etc.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/run.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* globals define */
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/run.js]: run.module.js

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js (hunk 1 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js]: > getting rid of $scope, private functions + vm.function, order of things
This is already fixed in #239 

I will fix the naming for controller and implement `$inject` in this PR.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+  'tasks-assignments/services/file',
+  'tasks-assignments/services/dialog',
+  'tasks-assignments/services/document',
+  'common/services/notification'
+], function (angular, moment, _, controllers) {
+  'use strict';
+
+  controllers.controller('ModalDocumentCtrl', ModalDocumentCtrl);
+
+  ModalDocumentCtrl.$inject = [
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: ModalDocumentCtrl.$inject = ['$scope', ...
  ..., 'notification'];

^ No need to break lines in this case.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+    'ContactService', 'FileService', 'data', 'files', 'config', 'HR_settings', 'modalMode',
+    'notification'
+  ];
+
+  function ModalDocumentCtrl ($scope, $modalInstance, $rootScope, $rootElement, $q, $log, role,
+    $filter, $modal, $dialog, $timeout, AssignmentService, DocumentService, ContactService,
+    FileService, data, files, config, HRSettings, modalMode, notification) {
+    $log.debug('Controller: ModalDocumentCtrl');
+
+      // Init call
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: No need in this (if not yet fixed in another PR)

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+    (function init () {
+      $scope.files = [];
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: Too much calls, need wrapping into funcs (if not yet fixed in another PR)

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+      $scope.dpOpened = {
+        due: false,
+        exp: false,
+        form: false
+      };
+
+      $scope.remindMeMessage = 'Checking this box sets a reminder that this document needs to be renewed a set number of days before the Expiry Date. You can set this by going <a target="_blank" href="/civicrm/tasksassignments/settings">here</a> CiviHR will do this by creating a copy of this document with the status ‘awaiting upload’, which you will be able to see in your Documents list.';
+    })();
+
+      /**
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: Something wrong with comment indentation across the whole file. Please fix if not yet fixed in another PR.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js (hunk 4 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: Fixed in PR #239

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js (hunk 6 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: Fixed in PR #239

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js (hunk 5 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: As most of the changes are already done in #239, I prefer to make changes you requested once #239 is merged.

**Comment by bmnepali** [comment]
Fixed Merge conflicts by rebase.

**Pull request merged**